Vendor engine-api to allow docker daemon reload event

yongtang opened the pull request:
This fix updated the vendored engine-api to version `e374c4fb5b121a8fd4295ec5eb91a8068c6304f4`, which defines a new event type of `DaemonEventType`. The purpose is to allow emitting `daemon reload` event as is raised in #22463 and #22590.

This fix is related to #22463 and #22590.

**NOTE:** In addition to update vendor/engine-api, this pull request also adds an additional commit to fix the failed test `TestRestartPolicy` due to the engine-api update.

After vendor/engine-api has been updated, the following unit test fails:

```
--- FAIL: TestRestartPolicy (0.00s)
       hostconfig_test.go:177: RestartPolicy.IsNone for { 0} should have been false but was true
```

See: https://jenkins.dockerproject.org/job/Docker-PRs/27408/console

The reason for the above failed unit test is that the pull request:

docker/engine-api#200

changed the behavior of the restart policy and makes `restartpolicy.IsNone` return true if restart policy name is `""`. As a result, the above mentioned unit test fails.

This second commit fixes the inconsistency of the unit test so that `TestRestartPolicy` could pass again.

Signed-off-by: Yong Tang yong.tang.github@outlook.com

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "22463-22590-emit-docker-daemon-reload-event" git@github.com:yongtang/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~2
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "22463-22590-emit-docker-daemon-reload-event" git@github.com:yongtang/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~2
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

yongtang committed:
Vendor engine-api to allow docker daemon reload event.

This fix updated the vendored engine-api to version
e374c4fb5b121a8fd4295ec5eb91a8068c6304f4, which defines a new event
type of `DaemonEventType`. The purpose is to allow emitting`
`daemon reload` event as is raised in #22463.

This fix is related to #22463 and #22590.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix failed test for TestRestartPolicy

This commit is a follow up of the last commit:
Vendor engine-api to allow docker daemon reload event.

After vendor/engine-api has been updated, the following
unit test fails:
```
--- FAIL: TestRestartPolicy (0.00s)
       hostconfig_test.go:177: RestartPolicy.IsNone for { 0} should have been false but was true
```

The reason for the above failed unit test is that pull request:

https://github.com/docker/engine-api/pull/200

updated behavior of the restart policy and makes restartpolicy.IsNone
return true if restart policy name is `""`. As a result, the above
mentioned unit test fails.

This fix fixes the inconsistency of the unit test so that `TestRestartPolicy`
could pass again.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

vdemeester commented:
Thanks @yongtang, LGTM
/cc @coolljt0725

coolljt0725 commented:
thanks @yongtang  LGTM
